xarray/core/accessor_str.py

```diff
@@ -883,6 +1105,42 @@ def count(
         Returns
         -------
         counts : array of int
+
+        Examples
+        --------
+        >>> da = xr.DataArray(["jjklmn", "opjjqrs", "t-euuJJvwx"], dims="x")
+        >>> da
+        <xarray.DataArray (x: 3)>
+        array(['jjklmn', 'opjjqrs', 't-euuJJvwx'], dtype='<U10')
+        Dimensions without coordinates: x
+
+        Using a string:
+        >>> da.str.count("jj")
+        <xarray.DataArray (x: 3)>
+        array([1, 1, 0])
+        Dimensions without coordinates: x
+
+        Enable case-insensitive matching by setting case to false:
+        >>> import re
+        >>> counts = da.str.count("jj", case=False)
+        >>> counts
+        <xarray.DataArray (x: 3)>
+        array([1, 1, 1])
+        Dimensions without coordinates: x
+
+        Using regex:
+        >>> counts = da.str.count(r"jj")
+        >>> counts
+        <xarray.DataArray (x: 3)>
+        array([1, 1, 0])
+        Dimensions without coordinates: x
+
+        Using a list of strings (returns a count of matches for each element):
+        >>> counts = da.str.count(["jj"])
```

Comment:
Not sure why this example returns values of 2, maybe I am misunderstanding something here...

Reply:
The docs mention that if the argument passed in is array-like, it is broadcast against the array and applied elementwise. So here the count function counts each occurrence of "j"  in the strings.

Reply:
That's not how I understand the text in the docstring.
Your list has length 1 and an element "jj", so from what I understand it should broadcast this to dimension "x", and then search element wise for "jj", so it should return an array of 1s again?

Reply:
Oooh... that makes more sense. I was misunderstanding broadcasting. Now, I don't get why it's returning 2s. 
When I pass in a data array, however (` pat = xr.DataArray(["jj"], dims="y")` ) instead of the list as the pattern, it returns 1s => `array([[1],[1],[0]])`

Reply:
Maybe you could make "y" a dimension of size two with different patters to highlight the broadcasting and element wise application.